### PR TITLE
DUPLO-17963 feat: add support for asynchronous retry features to duplocloud provider

### PR DESCRIPTION
**Pull request by Matheusbafutto**
## Summary of changes

This PR does the following:

- Adds new resource to manage Lambda's asynchronous invocation configuration

## Testing performed

- [ ] Using unit tests
- [x] Manually, on my local system
- [ ] Manually, on a remote test system

## Describe any breaking changes

No

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Added a new resource `duplocloud_aws_lambda_function_event_config` to manage AWS Lambda function event invoke configurations.
- Implemented CRUD operations for the new resource, including schema validation and API interactions.
- Updated the provider's resources map to include the new resource.
- Added documentation and examples for the new resource.
- Updated the changelog to reflect the addition of the new resource.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>provider.go</strong><dd><code>Register new AWS Lambda function event config resource</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

duplocloud/provider.go

<li>Added new resource <code>duplocloud_aws_lambda_function_event_config</code> to the <br>provider's resources map.<br> <li> Ensured the new resource is properly registered and available for use.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/595/files#diff-0959744faba70cfc67abd70733e39a6f668d41c9fae69a27968841ef8ed56eeb">+107/-106</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_aws_lambda_function_event_config.go</strong><dd><code>Implement AWS Lambda function event invoke config resource</code></dd></summary>
<hr>

duplocloud/resource_duplo_aws_lambda_function_event_config.go

<li>Created new resource file for managing AWS Lambda function event <br>invoke configurations.<br> <li> Implemented CRUD operations for the new resource.<br> <li> Defined schema and validation for the resource attributes.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/595/files#diff-7c2692b89eb57c436ee57c36d07de0ae15ba9ae2b4b15449d8f6fb59cb5c36ec">+256/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>aws_lambda_function.go</strong><dd><code>Add support for Lambda function event invoke config in SDK</code></dd></summary>
<hr>

duplosdk/aws_lambda_function.go

<li>Added new types for Lambda function event invoke configurations.<br> <li> Implemented API calls for creating, updating, retrieving, and deleting <br>event invoke configurations.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/595/files#diff-481201c30d6877a01e9dd350d11edce52e01e472cfab7e6b6ecaffca4ed8f6e2">+53/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>import.sh</strong><dd><code>Add import example for Lambda function event config</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/resources/duplocloud_aws_lambda_function_event_config/import.sh

<li>Added example script for importing AWS Lambda function event invoke <br>config.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/595/files#diff-fb7c298d951e3e350f8795ea77c6d653198c65ef0b5c02f9a2c4257d11d52258">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog with new Lambda function event config resource</code></dd></summary>
<hr>

CHANGELOG.md

<li>Documented the addition of the new AWS Lambda function event invoke <br>config resource.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/595/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+16/-10</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>aws_lambda_function_event_config.md</strong><dd><code>Document new AWS Lambda function event config resource</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/resources/aws_lambda_function_event_config.md

<li>Added documentation for the new <br><code>duplocloud_aws_lambda_function_event_config</code> resource.<br> <li> Included example usage and schema details.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/595/files#diff-948452f324535328db25e85598cee64bf1e0212ae15539349d97005031e32850">+144/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource.tf</strong><dd><code>Add example Terraform config for Lambda function event config</code></dd></summary>
<hr>

examples/resources/duplocloud_aws_lambda_function_event_config/resource.tf

<li>Provided example Terraform configuration for the new AWS Lambda <br>function event invoke config resource.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/595/files#diff-731bdcf299f512c34d27ee7f7a0c9c8b7cdf48c693749449e1a26f28f0b09552">+64/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-17963 TF:Add support for lambda asynchronous retries](https://app.clickup.com/t/8655600/DUPLO-17963)

**Comment by Matheusbafutto** [comment]
Merge after https://github.com/duplocloud-internal/duplo/pull/2921

File: duplocloud/resource_duplo_aws_lambda_function_event_config.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+				},
+			},
+		},
+	}
+}
+
+// Resource for managing an AWS lambda function
+func resourceAwsLambdaEventInvokeConfigFunction() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_aws_lambda_function` manages an AWS lambda function in Duplo.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_aws_lambda_function_event_config.go:98]: Change description from `duplocloud_aws_lambda_function` to `duplocloud_aws_lambda_function_event_config`

**Comment by tahir-duplo** [comment]
Please add an example as well along with the import command.

File: duplocloud/resource_duplo_aws_lambda_function_event_config.go (hunk 1 above)
Comment (Matheusbafutto) [reply duplocloud/resource_duplo_aws_lambda_function_event_config.go:98]: done

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Review by komalkadam-dc — APPROVED** [review]

**Pull request merged**